README.md

```diff
@@ -0,0 +1,684 @@
+    crud/read --> crud/*
+    crud/mutate --> crud/*
+
+    subgraph mutationGraph [ ]
+        crud/create --> crud/mutate
+        crud/update --> crud/mutate
+        crud/destroy --> crud/mutate
+    end
+  end
+
+  ... --> *
+```
+
+## 4.4 Caveats
+
+Caveats define a set of constraints on what can be re-delegated or invoked. Caveat semantics MUST be established by the Subject. They are openly extensible, but vocabularies may be reused across many Subjects. Caveats MUST be Understood by the [Executor] of the eventual [Invocation][UCAN Invocation]. Each caveats MUST be formatted as a map.
+
+``` js
+{
+  "sub": "did:web:example.com",
+  "can": "crud/update",
+  "iff": [
+     {                                    // ┐
+       "uri": "https://blog.example.com", // │
+       "status": "published"              // ├─ Caveat
+       "tag": "news",                     // │
+     },                                   // ┘
+     {                   // ┐
+       "tag": "breaking" // ├─ Caveat
+     }                   // ┘
+  ],
+  // ...
+}
+```
+
+The above MUST be interpreted as the set of capabilities below in the following table:
+
+| Subject               | Ability       | Caveat                                                                                                 |
+|-----------------------|---------------|--------------------------------------------------------------------------------------------------------|
+| `did:web:example.com` | `crud/update` | Posts at `https://blog.example.com` with the `published` status and tagged with `news` and `breaking`  |
+
+The `iff` caveat set MUST take the following shape: `[{}]`. The arrays represents logical `all` (chained `AND`s). To represent logical `any` / `OR`, issue another delegation with that attenuation. For instance, the following represents `{a: 1, b:2} AND {c: 3} AND {d: 4}`:
+
+``` js
+[
+  {       // ┐
+    a: 1, // ├─ Caveat ─┐
+    b: 2  // │          │
+  },      // ┘          ├─ AND ─┐
+  {       // ┐          │       │
+    c: 3  // ├─ Caveat ─┘       │
+  },      // ┘                  ├─ AND
+  {       // ┐                  │
+    d: 4  // ├─ Caveat ─────────┘
+  }       // ┘
+]
+```
+
+Expressing caveats in this standard way simplifies ad hoc extension at delegation time. As a concrete example, if a root UCAN caveat has a `tag` field but no (plural) `tags` field, it is still possible to express multiplicity by adding another `AND`ed caveat:
+
+``` js
+// Original Caveat
+[
+  {
+    "uri": "https://blog.example.com/",
+    "tag": "news"
+  }
+]
+
+// Attenuated Caveat
+[
+  {
+    "uri": "https://blog.example.com/",
+    "tag": "news"
+  }, 
+  // AND
+  {
+    "tag": "breaking"
+  }
+]
+```
+
+In this example, the original caveat had not accounted for there being multiple tags at runtime. The attenuated capability grants access to blog posts at `https://blog.example.com/` that are tagged with both `news` and `breaking` due to the `AND` in the predicate.
+
```

Comment:
Personally, I'd love for caveats to simply be arbitrary metadata.
If a service starts out with a `{ "tag": "something" }` caveat, then it can eventually upgrade to `{ "tags": [...] }` by parsing the old format into the new.
I want libraries like rs-ucan to be dumb. They should not try to figure out how caveats subsume or not subsume each other nor should the UCAN spec define *the* way of how caveats subsume each other.

Caveat subsumption can be complex. It's not far-fetched to think that a use case may want to do complex cryptography, or perhaps (as in the ucanto RPC case) a full expression language on a level similar to SQL.
At the same time there will be use-cases that just don't need any of this.

As an implementer of a service, I wouldn't mind having to provide the exact caveat subsumption logic to rs-ucan (and perhaps being able to pick a simple default one that's similar to the one described in this spec).
However, I'd be unhappy if I'm forced to specify my caveats in exactly the format that UCAN delegations define, while a simpler service-specific definition would work.

The other side of the coin may be interoperability, but I don't see how yet.
We've always had "services describe their own semantics for delegation", and that's worked out? (Right?)

Reply:
Yeah I agree with @matheus23. Historically UCAN has been heavy on semantics and light on syntactic rules. I'm increasingly convinced there where we can make things syntactic, the libraries can become a lot "dumber". There's still a plenty of semantics to contend with, but we can reduce that surface area significantly.